launchable/utils/launchable_client.py

```diff
@@ -102,15 +102,29 @@ def print_exception_and_recover(self, e: Exception, warning: Optional[str] = Non
             click.echo(click.style(warning, fg=warning_color), err=True)
 
     def is_fail_fast_mode(self) -> bool:
-        if 'fail_fast_mode' in self._workspace_state_cache:
-            return self._workspace_state_cache['fail_fast_mode']
-        # TODO: call api and set the result to cache
```

Comment:
Would it be possible to use https://docs.python.org/3/library/functools.html#functools.cached_property?